python/cuml/test/experimental/test_explainer_shap.py

```diff
@@ -0,0 +1,180 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cuml
+import cuml.experimental.explainer
+import numpy as np
+import pytest
+
+from cuml.common.import_utils import has_shap
+from cuml.test.utils import array_equal
+from cuml.test.utils import ClassEnumerator
+from sklearn.datasets import make_classification
+from sklearn.model_selection import train_test_split
+
+
+models_config = ClassEnumerator(module=cuml)
+models = models_config.get_models()
+
+golden_results = {
+    (4, cuml.LinearRegression): [58.13167305, 139.33765425, 28.08136872,
+                                 13.12541971],
+    (10, cuml.LinearRegression): [-3.47197726, -12.13657959, -43.05540892,
+                                  -41.44955195, -4.1909009, -30.91657623,
+                                  -14.73675613, 23.92447365, 15.73265123,
+                                  -45.94585396],
+    (4, cuml.KNeighborsRegressor): [58.13167305, 139.33765425, 28.08136872,
+                                    13.12541971],
+    (10, cuml.KNeighborsRegressor): [-3.47197726, -12.13657959, -43.05540892,
+                                     -41.44955195, -4.1909009, -30.91657623,
+                                     -14.73675613, 23.92447365, 15.73265123,
+                                     -45.94585396]
+}
+
+
+# todo: use tags to generate the correct dataset
+@pytest.fixture(scope="session")
+def single_dataset():
+    X, y = make_classification(100, 5, random_state=42)
+    X = X.astype(np.float32)
+    y = y.astype(np.float32)
+    return X, y
```

Comment:
If this is something that should be used session wide, it should be moved to a `conftest.py`. Otherwise, this should be either "module" or some more narrow scope. Otherwise it will keep the memory in use for the rest of the pytest run.